src/main/java/sqlline/SqlLineOpts.java

```diff
@@ -779,6 +796,23 @@ private String getValidDateTimePatternOrThrow(String dateTimePattern) {
     return dateTimePattern;
   }
 
+  private String getValidConfirmPatternOrThrow(String confirmPattern) {
+    if (confirmPattern == null) {
+      throw new IllegalArgumentException("Confirm pattern is null");
+    }
+    if (DEFAULT.equalsIgnoreCase(confirmPattern)) {
+      confirmPattern = sqlLine.loc("default-confirm-pattern");
+    }
+    set(CONFIRM_PATTERN, confirmPattern);
+    try {
+      Pattern.compile(confirmPattern);
```

Comment:
Creating a new Pattern object everytime is expensive (good to have optimization)

Refer: https://stackoverflow.com/a/1721778/4428999
Cache this object and reuse when checking.

Reply:
Thank you for the suggestion @karanmehta93